extensions/applicationinsights-cfgsync-js/src/CfgSyncHelperFuncs.ts

```diff
@@ -39,3 +42,82 @@ export function replaceByNonOverrideCfg<T=IConfiguration & IConfig, T1=NonOverri
     // if errors happen, do nothing
     return cfg;
 }
+
+//                                cdn Mode (cdn Value = B, SDK Config Defaults = C)
+//                   |--------------------------------------------------------------------------|
+//                   |                    | none        | disabled    | enabled     | force     |
+//                   | ------------------ | ----------- | ----------- | ----------- | --------- |
+// | User Mode       | none               | A || C      | B || A || C | B || A || C | B || C    |
+// (user Value = A)  | disabled           | A || C      | B || A || C | A || C      | B || C    |
+//                   | enabled            | A || C      | B || A || C | A || B || C | B || C    |
+//                   | none(blockCdn)     | A || C      | A || C      | A || C      | A || C    |
+//                   | disabled(blockCdn) | A || C      | A || C      | A || C      | A || C    |
+//                   | enabled(blockCdn)  | A || C      | A || C      | A || C      | A || C    |
+export function shouldOptInFeature(field: string, cdnCfg?: ICfgSyncCdnConfig, customOptInDetails?: IFeatureOptIn) {
+    
+    if (!cdnCfg || !cdnCfg.enabled || !cdnCfg.featureOptIn || !cdnCfg.featureOptIn[field] || !customOptInDetails || !customOptInDetails[field]) {
+        return null;
+    }
+ 
+    let cdnFeature = cdnCfg.featureOptIn[field];
+    let cdnMode = cdnFeature.mode || CdnFeatureMode.none;
+    let cdnFeatureVal = cdnFeature.value;
+    let customFeature = customOptInDetails[field];
+    let customMode = customFeature.mode || FeatureOptInMode.disable; // default custom mode is disable
+    let customFeatureVal = customFeature.cfgValue;
+    let blockCdn = customFeature.blockCdnCfg || false;
+
+    if (blockCdn) {
+        return customFeatureVal;
+    }
+
+    if (cdnMode === CdnFeatureMode.force) {
+        return cdnFeatureVal;
+    }
+    let featureVal = customFeatureVal;
+    if (cdnMode === CdnFeatureMode.disable) {
+        if (!isNullOrUndefined(cdnFeatureVal)) {
+            featureVal = cdnFeatureVal;
```

Comment:
We could probably just return here and avoid the extra check for the `cdnMode === CdnFeatureMode.enable` on 84 and 90